### PR TITLE
[raylib] Add remaining headers and fix CMake

### PR DESCRIPTION
**Pull request by dhkatz**
**Describe the pull request**

- What does your PR fix? Two headers were missing from raylib: raymath.h and rlgl.h. While these headers are NOT required to use the base library, they are expected headers to use in tandem with, or standalone to raylib.

- Which triplets are supported/not supported? Have you updated the CI baseline? Should support all the same triplets.

- Does your PR follow the [maintainer guide](https://github.com/microsoft/vcpkg/blob/master/docs/maintainers/maintainer-guide.md)?

### CODE REVIEW

**Comment by dhkatz** [comment]
In raylib's own cmake files, a couple other headers seem to also be intended to be copied over as well:

```cmake
# Copy the header files to the build directory
file(COPY "raylib.h" DESTINATION ".")
file(COPY "rlgl.h" DESTINATION ".")
file(COPY "physac.h" DESTINATION ".")
file(COPY "raymath.h" DESTINATION ".")
file(COPY "raudio.h" DESTINATION ".")
```

**Comment by dhkatz** [comment]
Ok I don't know what happened but all of a sudden find_package fails for raylib? With my changes and with what is on master. I even tried completely reinstalling vcpkg and find_package still fails to find raylib. All the other packages I use work.

```
CMake Error at C:/Program Files/JetBrains/apps/CLion/ch-0/201.7223.86/bin/cmake/win/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
  Could NOT find raylib (missing: raylib_LIBRARY)
Call Stack (most recent call first):
  C:/Program Files/JetBrains/apps/CLion/ch-0/201.7223.86/bin/cmake/win/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
  C:/Users/mcilu/vcpkg/installed/x64-windows-static/share/raylib/raylib-config.cmake:48 (find_package_handle_standard_args)
  C:/Users/mcilu/vcpkg/installed/x64-windows-static/share/raylib/vcpkg-cmake-wrapper.cmake:3 (_find_package)
  C:/Users/mcilu/vcpkg/scripts/buildsystems/vcpkg.cmake:284 (include)
  CMakeLists.txt:57 (find_package)
```

**Comment by LilyWangL** [comment]
> Ok I don't know what happened but all of a sudden find_package fails for raylib? With my changes and with what is on master. I even tried completely reinstalling vcpkg and find_package still fails to find raylib. All the other packages I use work.
> 
> ```
> CMake Error at C:/Program Files/JetBrains/apps/CLion/ch-0/201.7223.86/bin/cmake/win/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:146 (message):
>   Could NOT find raylib (missing: raylib_LIBRARY)
> Call Stack (most recent call first):
>   C:/Program Files/JetBrains/apps/CLion/ch-0/201.7223.86/bin/cmake/win/share/cmake-3.16/Modules/FindPackageHandleStandardArgs.cmake:393 (_FPHSA_FAILURE_MESSAGE)
>   C:/Users/mcilu/vcpkg/installed/x64-windows-static/share/raylib/raylib-config.cmake:48 (find_package_handle_standard_args)
>   C:/Users/mcilu/vcpkg/installed/x64-windows-static/share/raylib/vcpkg-cmake-wrapper.cmake:3 (_find_package)
>   C:/Users/mcilu/vcpkg/scripts/buildsystems/vcpkg.cmake:284 (include)
>   CMakeLists.txt:57 (find_package)
> ```

Did you run `.\vcpkg.exe integrate install` on VCPKG?

**Comment by dhkatz** [comment]
@LilyWangL With the help of @qis on Slack it turns it this is actually a bug with the port. The reason it started failing suddenly is that I switched my default triplet from `x64-windows` to `x64-windows-static`, and there is a bug in the `raylib-config.cmake` file which prevents it from seeing the static version of the library. I will include this fix in my PR.

**Review by LilyWangL — APPROVED** [review]

File: ports/raylib/fix-cmake-static.patch [hunk 1]
```diff
@@ -0,0 +1,17 @@
...
+@@ -27,11 +27,7 @@ find_path(raylib_INCLUDE_DIR
+     HINTS ${${XPREFIX}_INCLUDE_DIRS}
+ )
+ 
+-set(RAYLIB_NAMES raylib)
+-
+-if (raylib_USE_STATIC_LIBS)
+-    set(RAYLIB_NAMES libraylib.a raylib.lib ${RAYLIB_NAMES})
+-endif()
++set(RAYLIB_NAMES raylib raylib_static)
```
Comment (dan-shaw) [inline ports/raylib/fix-cmake-static.patch:14]: We should use `vcpkg_cmake_wrapper.cmake` to rename the static and shared targets. See the library `rocksdb` for an example. Also see #6045

**Comment by LilyWangL** [comment]
Pinging @dhkatz

**Comment by LilyWangL** [comment]
Pinging @dhkatz for response. Is work still being done for this PR?

**Comment by LilyWangL** [comment]
Closing this PR since it seems that no progress is being made. Please reopen if work is still being done.

**Pull request closed without merging**